src/python/pants/help/scope_info_iterator.py

```diff
@@ -1,23 +1,23 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Set, Type
 
 from pants.option.global_options import GlobalOptionsRegistrar
 from pants.option.parser_hierarchy import enclosing_scope
 from pants.option.scope import GLOBAL_SCOPE, ScopeInfo
 from pants.subsystem.subsystem_client_mixin import SubsystemClientMixin
 
 
+@dataclass
```

Comment:
Yay! Thanks! Is it possible to make this `frozen=True`? Even if this isn't used by the engine, it's a sane default.

You'll still be able to `self.scope_infos.extend()`, i.e. mutate the values of the fields, only can't do `self.scope_infos = []`.